### DISCUSSION TITLE
TableView setOnMouseClicked

### DISCUSSION THREAD

**Discussion by julianprimetshofer** [palexdev/MaterialFX#237] [Archived]
Hey,

I got another problem. Currently I'm using the MFXTableView and I want to add the possibility to double-click on a table row. Normally I would do it like this:

```java
table.setRowFactory(tv -> {
    TableRow<MyType> row = new TableRow<>();
    row.setOnMouseClicked(event -> {
        if (event.getClickCount() == 2) {
            MyType rowData = row.getItem();
            System.out.println(rowData);
        }
    });
    return row ;
});
```

But somehow it looks like this is no working for the MFXTableView. I appreciate any help!

**Comment by palexdev** [comment]
Maybe there's something intercepting and consuming the event
Did you fix it?
If not you could try to do row.addEventFilter(MouseEvent.MOUSE_CLICKED, event -> {...});

**Reply by julianprimetshofer** [reply]
This fixed it. Thanks for your help

**Reply by foxxbl** [reply]
This also helped me, thanks!!
Apparently setOnMouseClicked doesn't work with MFXTableView.
